client/utils/index.js

```diff
@@ -8,3 +8,13 @@ export const getAddToCartVariationParams = ( key ) => {
 
 	return wcAddToCartVariationParams[ key ];
 };
+
+/**
+ * Check if APMs are deprecated. Meaning it is past Oct 31st, 2024 and the legacy checkout is enabled.
+ *
+ * @return {boolean} Whether APMs are deprecated.
+ */
+export const areAPMsDeprecated = () =>
+	new Date() > new Date( '2024-10-31' ) &&
```

Comment:
It should be Oct 29th (the same date mentioned in banner message)
```suggestion
	new Date() > new Date( '2024-10-29' ) &&
```

I like @annemirasol suggestion to rely on the server for this flag; that way, we don't need to worry about the customer's browser date being wrong 🤷🏼